packages/application-shell/src/components/navbar/devonly-menu-loader.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { wrapDisplayName } from 'recompose';
+
+const noop = () => Component => Component;
+
+const devonlyMenuLoader =
+  process.env.NODE_ENV !== 'development'
```

Comment:
Dealing with loading the `menu.json` config for development was/is tricky to do, not much for the implementation but for the **approach**.

For instance, on production the application uses the `https://mc.commercetools.com/api/graphql` endpoint to load the configs through the Proxy server. **In development though, there is no proxy running**.
Furthermore, you want to load the `menu.json` of the application you're **developing from**.
In fact, you can't use the links to the other apps because they lead to 404 pages.

With that in mind, we need to **disable** the graphql queries to the proxy endpoint, since the URL is defined as `${window.location.origin}/api/graphql`, which in development mode points to `http://localhost:3001/api/graphql`.
I thought about having a fake graphql endpoint in the webpack dev server, but there are some disadvantages with that:
- it's tight to webpack (e.g. if someone wants to develop an app using something different, there is a problem)
- we have duplication of the graphql schema in two places (app-kit + mc proxy)

I also tried using a custom apollo link to write to the cache with based on the `menu.json`, so that no network request is made since apollo will try to read from the cache. However, besides the implementation being a bit complex, it seems that apollo makes the request anyway, probably because the cache is not being filled in time.

This leaves us with the final approach I took, which is maybe not the ideal solution but it's good enough. One of the important things to keep in mind was that the setup should be as simple as possible for the user:

```js
DEV_ONLY__getNavbarMenuConfig={() => import('../../../menu.json')}
```

I chose to explicitly mark the prop with `DEV_ONLY_` to make it clear that this won't be used in production.

Little downside: I suppose webpack will create a chunk for the `menu.json`, since it finds `import()` split point. We could e.g. wrap the call with a `process.env.NODE_ENV` so that uglily will remove it, but I'm not sure if that makes things simpler.

Please let me know what do you think about this approach, about naming etc. I'm open to suggestions 😇

Reply:
yeah maybe wrapping the call would be a good idea? but another unused chunk in prod doesn't really make any difference 🤔

Reply:
Could we also just have a HoC which reads the process.env to alternate the fetching behaviour? So instead of having a `withDevOnlyMenuLoader` and some `graphql` (Apollo) application we have a `withMenuQuery` and it alternates between fetching from the .json and through Apollo based on the env or a specific prop?

Reply:
Yeah, I think @montezume and I mean the same. Wrap the call in the HoC itself.

Reply:
Sure, I can try that